utils/jsonServerFunctions.ts

```diff
@@ -1,45 +1,51 @@
 import axios from 'axios';
+import { Gifts } from '~/pages';
 
-const baseURL = 'http://localhost:3001/gifts';
+//const baseURL = 'http://localhost:3001/gifts'; // localhost url
+const baseURL = 'https://my-json-server.typicode.com/samuliasmala/lahjalista/gifts'; // external url
 
-async function getAll() {
-  return await axios.get(baseURL);
+/**
+ *
+ * @returns an array that contains all the gifts as objects
+ */
+export async function getAllGifts() {
+  return (await axios.get(baseURL)).data as Gifts[];
 }
 
 /**
  *
- * @param searchParameters should be given a string that contains search parameters as query strings.
- * @example name=Foo&gift=Bar
- * @returns an array of JavaScript objects if found. Else an empty array
+ * @param id should be given a string that contains the id that is wanted to be searched.
+ * @returns an object of a gift if it was found. Else it will return null
  */
-async function getOne(searchParameters: string) {
-  return (await axios.get(`${baseURL}?${searchParameters}`)).data;
+export async function getGift(id: string) {
+  const gifts = (await axios.get(`${baseURL}?id=${id}`)).data;
+  if (gifts.length != 0) {
+    return gifts[0];
```

Comment:
Kannattaa kertoa oikea tyyppi (oletetaan että backend palauttaa lahjan kuten pitää)
```suggestion
    return gifts[0] as Gifts;
```